libgrust/libproc_macro/rust/bridge/ffistring.rs

```diff
@@ -0,0 +1,48 @@
+use std::convert::TryInto;
+use std::ffi::c_uchar;
+use std::fmt;
+use std::slice::from_raw_parts;
+use std::str::from_utf8;
+
+extern "C" {
+    fn FFIString__new(data: *const c_uchar, len: u64) -> FFIString;
+    fn FFIString__drop(string: *mut FFIString);
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct FFIString {
+    data: *const c_uchar,
+    len: u64,
+}
+
+impl FFIString {
+    pub fn new(string: &str) -> FFIString {
```

Comment:
I think `clippy` will warn here, won't it? `Self::new` should indicate the creation of an empty `FFIString`, so equivalent to what you have right now with `FFIString::new("")`, and we should instead implement `From<&str> for FFIString`? What do you think?

Reply:
If there would be any trait here to use that would probably be [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) instead of `From<T>`. I'm not a fan of importing traits or any rust mechanism that is not strictly required in this particular crate as we want to compile it early on. Using more rust things means it may not compile because the compiler misses a bit somewhere that we do not have implemented yet. I mean,  this could be changed later once the compiler is working "better". What are your thoughts about this ?

Reply:
We do use FromStr in this crate, thus I was enclined to change this code, but I then remembered it returned a Result. I therefore implemented the `From<S: AsRef<str>>` instead in [f5f749b](https://github.com/Rust-GCC/gccrs/pull/2204/commits/f5f749b1fb415ea23fdf7be559970fa04909b30f).